pkg/commands/git_commands/working_tree.go

```diff
@@ -46,10 +47,9 @@ func (self *WorkingTreeCommands) StageFile(path string) error {
 }
 
 func (self *WorkingTreeCommands) StageFiles(paths []string) error {
-	quotedPaths := make([]string, len(paths))
-	for i, path := range paths {
-		quotedPaths[i] = self.cmd.Quote(path)
-	}
+	quotedPaths := slices.Map(paths, func(path string) string {
+		return self.cmd.Quote(path)
+	})
```

Comment:
most of the changes in this PR are things like this. It ends being roughly as many lines but it's good not have to worry about choosing between `make([]string, len(paths))` and `make([]string, 0, len(paths))` and the possible errors that arise from confusing the two.

I find that the `func(path string) string` part of this example can easily be ignored so that you're just paying attention to the fact we're mapping over paths to get their quotes.

One benefit of using .Map over a for loop (and same deal with all the other methods here) is that you're told upfront what's happening. You don't need to read all the way through a for loop to know that we're doing a map. This is useful when debugging because often you can skip over whole chunks of code if you know that for example the length of a slice won't change within a map.